tests/ProjectsMetadataTest.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ManageApiTest;
+
+use Keboola\ManageApi\Client;
+use Keboola\ManageApi\ClientException;
+use Keboola\ManageApi\ProjectRole;
+
+class ProjectsMetadataTest extends ClientTestCase
+{
+    public const TEST_METADATA = [
+        [
+            'key' => 'test_metadata_key1',
+            'value' => 'testval',
+        ],
+    ];
+
+    public const PROVIDER_USER = 'user';
+
+    private $organization;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->client->addUserToMaintainer($this->testMaintainerId, ['email' => 'spam+spam@keboola.com']);
+
+        foreach ($this->client->listMaintainerMembers($this->testMaintainerId) as $member) {
+            if ($member['id'] === $this->normalUser['id']) {
+                $this->client->removeUserFromMaintainer($this->testMaintainerId, $member['id']);
+            }
+
+            if ($member['id'] === $this->superAdmin['id']) {
+                $this->client->removeUserFromMaintainer($this->testMaintainerId, $member['id']);
+            }
+        }
+
+        $this->organization = $this->client->createOrganization($this->testMaintainerId, [
+            'name' => 'My org',
+        ]);
+
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUserWithMfa['email']]);
+        $this->client->removeUserFromOrganization($this->organization['id'], $this->superAdmin['id']);
+    }
+
+    public function testNormalUserCannotAddMetadata(): void
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->superAdmin['email']]);
+        $projectId = $this->createProjectWithSuperAdminMember($this->organization['id']);
+
+        try {
+            $this->normalUserClient->setProjectMetadata($projectId, self::PROVIDER_USER, self::TEST_METADATA);
+            $this->fail('Should fail.');
+        } catch (ClientException $e) {
+            $this->assertEquals(403, $e->getCode());
+        }
+    }
+
+    public function testSuperAdminCanAddMetadataInOrgWithAllowAutoJoinFalse()
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUser['email']]);
```

Comment:
Jak se tady furt prohazuje kdo je a kdo není kde, tak je to hrozný zmatek :'( Skoro by to chtělo nějakého "test setup usera", který není součástí testů a dělá se s ním tohle všechno. A k tomu mít "testovacího superusera" na kterém se budou dělat ty testy. Tohle je 🤯

Reply:
No imho ak by si pridal aj toho testovacieho usera, tak by si musel robit podobne komba a to z dovodu, ze nezarucis aby mu niekto nahodou nejak omylom nenastavil nejake prava co nema mat, ano mozes to nejak kontrolovat na review ale na to by som sa nespoliehal, preto je podla man spraveny `ProjectInvitationTest` takto a ja som z toho cerpal.

Mne to ako zmatok zas tak moc nepride:
1. v setUp si zmazes normal user a superAdmina z maintainera
2. vytvoris novu org a odoberies super admina ktorym si to zkladal aby si ho nemal v org
3. Mas teda normalnUser a super user ktory niesu sucastion maintanera a novu org (ked na to teraz este pozeram tak je asi chyba tam pridavat toho normalUserWithMfa)

A teraz chces otestovat ze ak si super admin tak mozes pridat metadata aj ked `allowedAutoJoin=false`
1. potrebujes novy projekt, takze si mozes normalUserom zalozit projekt pretoze vies ze ho v tom teste nepotrebujes, takze ho pridas do tej org a zalozis novy projekt
2. updatnes tu org a 'allowAutoJoin' => false,
3. otestujes ze super admin moze pridat metadata